### ISSUE TITLE
Install Errors

### ISSUE THREAD

**Issue by rwhitmire** [smsohan/MvcMailer#33]
on installation from within a brand new MVC3 C# project, I get the following errors:

PM> install-package MvcMailer
Attempting to resolve dependency 'T4Scaffolding (≥ 0.9.7)'.
Attempting to resolve dependency 'EntityFramework (≥ 4.1.10311.0)'.
You are downloading EntityFramework from Microsoft, the license agreement to which is available at http://go.microsoft.com/fwlink/?LinkId=224682. Check the package for additional dependencies, which may come with their own license agreement(s). Your use of the package and dependencies constitutes your acceptance of their license agreements. If you do not accept the license agreement(s), then delete the relevant components from your device.
Successfully installed 'EntityFramework 4.1.10715.0'.
Successfully installed 'T4Scaffolding 0.9.9'.
Successfully installed 'MvcMailer 1.1'.
Successfully removed 'EntityFramework 4.1.10331.0' from MvcApplication2.
Successfully added 'EntityFramework 4.1.10715.0' to MvcApplication2.
Successfully added 'T4Scaffolding 0.9.9' to MvcApplication2.
Set-DefaultScaffolder : Unable to cast object of type 'NuGet.SemanticVersion' to type 'System.Version'.
At C:\users\rwhitmire\documents\visual studio 2010\Projects\MvcApplication2\packages\T4Scaffolding.0.9.9\tools\install.ps1:22 char:22
- Set-DefaultScaffolder <<<<  -Name DbContext -Scaffolder T4Scaffolding.EFDbContext -SolutionWide -DoNotOverwriteExistingSetting
  - CategoryInfo          : NotSpecified: (:) [Set-DefaultScaffolder], InvalidCastException
  - FullyQualifiedErrorId : T4Scaffolding.Cmdlets.SetDefaultScaffolderCmdlet

Set-DefaultScaffolder : Unable to cast object of type 'NuGet.SemanticVersion' to type 'System.Version'.
At C:\users\rwhitmire\documents\visual studio 2010\Projects\MvcApplication2\packages\T4Scaffolding.0.9.9\tools\install.ps1:23 char:22
- Set-DefaultScaffolder <<<<  -Name Repository -Scaffolder T4Scaffolding.EFRepository -SolutionWide -DoNotOverwriteExistingSetting
  - CategoryInfo          : NotSpecified: (:) [Set-DefaultScaffolder], InvalidCastException
  - FullyQualifiedErrorId : T4Scaffolding.Cmdlets.SetDefaultScaffolderCmdlet

Set-DefaultScaffolder : Unable to cast object of type 'NuGet.SemanticVersion' to type 'System.Version'.
At C:\users\rwhitmire\documents\visual studio 2010\Projects\MvcApplication2\packages\T4Scaffolding.0.9.9\tools\install.ps1:24 char:22
- Set-DefaultScaffolder <<<<  -Name CustomTemplate -Scaffolder T4Scaffolding.CustomTemplate -SolutionWide -DoNotOverwriteExistingSetting
  - CategoryInfo          : NotSpecified: (:) [Set-DefaultScaffolder], InvalidCastException
  - FullyQualifiedErrorId : T4Scaffolding.Cmdlets.SetDefaultScaffolderCmdlet

Set-DefaultScaffolder : Unable to cast object of type 'NuGet.SemanticVersion' to type 'System.Version'.
At C:\users\rwhitmire\documents\visual studio 2010\Projects\MvcApplication2\packages\T4Scaffolding.0.9.9\tools\install.ps1:25 char:22
- Set-DefaultScaffolder <<<<  -Name CustomScaffolder -Scaffolder T4Scaffolding.CustomScaffolder -SolutionWide -DoNotOverwriteExistingSetting
  - CategoryInfo          : NotSpecified: (:) [Set-DefaultScaffolder], InvalidCastException
  - FullyQualifiedErrorId : T4Scaffolding.Cmdlets.SetDefaultScaffolderCmdlet

Successfully added 'MvcMailer 1.1' to MvcApplication2.
Set-DefaultScaffolder : Unable to cast object of type 'NuGet.SemanticVersion' to type 'System.Version'.
At C:\users\rwhitmire\documents\visual studio 2010\Projects\MvcApplication2\packages\MvcMailer.1.1\tools\install.ps1:23 char:22
- Set-DefaultScaffolder <<<<  -Name Mailer -Scaffolder $mailerScaffolder -SolutionWide -DoNotOverwriteExistingSetting
  - CategoryInfo          : NotSpecified: (:) [Set-DefaultScaffolder], InvalidCastException
  - FullyQualifiedErrorId : T4Scaffolding.Cmdlets.SetDefaultScaffolderCmdlet

**Comment by smsohan** [comment]
Thanks for reporting this. I will have to try this when I get some
free time in this week. Can you try the same thing on a fresh project?
In the past I have seen people having such issues when a previously
installed Nuget package was removed but the dependencies were still

## hanging around in the project.

Sohan
http://smsohan.com
skype:smsohan | gtalk:sohan39 | cell: 403-714-2673

On Fri, Dec 30, 2011 at 12:54 AM, Ryan Whitmire
reply@reply.github.com
wrote:

**Comment by rwhitmire** [comment]
yes, this occurs on a fresh project, both MVC3 and Empty Web Application.

**Comment by comfreakph** [comment]
try to Install separately the T4Scaffolding

**Comment by drgrieve** [comment]
I had the same issue. As comfreakph mentions you MUST install T4scaffolding first. (and possibly restart VS) before installing MvcMailer.

**Comment by rwhitmire** [comment]
OK thanks!  I'm writing an app that need some email stuff now so I'll try it out in the next day or two.

**Comment by krokonoster** [comment]
Just to confirm, after struggling the better part of an hour to get this working:
I had the same issues, so installed T4scaffolding, restart VS and then installed MvcMailer.
Ran T4scaffolding "Scaffold Mailer UserMailer Welcome,GoodBye" and FINALLY it's working.
Too nice a tool this to give up on, but seriously try and sort out this issues.  :-)

**Comment by ericjohannsen** [comment]
Same issue here.  Hopefully it can be resolved soon.  Great package, hate to lose people that can't get past this problem.

**Comment by ctolkien** [comment]
Hit this issue as well...

**Comment by ericjohannsen** [comment]
All issues I hit are covered (with solutions) here:

http://stackoverflow.com/questions/10241797/error-scaffolding-with-mvcmailer-in-mvc-4/10309336#10309336

**Comment by xaviguardia** [comment]
Still fails

**Comment by xaviguardia** [comment]
Works fine if you install first mvcscaffolding & then mvcmailer

**Comment by gluip** [comment]
Same here. Took me 30 minutes to find this thread..please fix

**Comment by shoaibshakeel381** [comment]
when is this going to be fixed??
